### PR TITLE
Improve the Device type id handling

### PR DESCRIPTION
**Pull request by mutesplash**
This patch centralizes the device type id handling by moving all the magic numbers into one place.  This improves code readability for when exceptions are made for specific device types (such as Matrix).  It also simplifies adding new devices supported by existing code.  Finally, since the BuildHAT is capable of distinguishing between different types of devices such as motors, a description section was added to Hat.get() to aid in determining which device is connected.

### CODE REVIEW

**Comment by chrisruk** [comment]
Thanks for this, and other pull requests :) I'll have a look at testing this soon

File: buildhat/devicetypes.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                      63: "Force Sensor", #45606
+                      64: "3x3 Color Light Matrix", #45608
+                      38: "Medium Linear Motor", # 88008
+                      46: "Large Motor", #88013
+                      47: "XL Motor", #88014
+                      48: "Medium Angular Motor (Cyan)", #45603 RPi docs says this is the "Large Angular" (Cyan), which is an emperical mismatch against 45603
+                      49: "Motor", # BuildHAT serial protocol PDF says this is the "SPIKE Prime large motor" (which is Cyan).  RPi docs call id 31 "Medium Angular Motor"
+                      65: "Small Angular Motor", #45607
+                      75: "Medium Angular Motor", # Maybe 88018, came with 51515, which lists as 6299646, not 6359216
+                      76: "Motor"
```
Comment (dlech) [inline buildhat/devicetypes.py:40]: You might find [this](https://github.com/pybricks/technical-info/blob/master/assigned-numbers.md#io-device-type-ids) useful to solve some of the ?s here.

Comment (mutesplash) [reply buildhat/devicetypes.py]: Indeed.  The Mario codes are fascinating in that they're in the same space, but over bluetooth.

**Comment by mutesplash** [comment]
Thanks for accepting them!  This code could probably be improved by using an enum instead of string comparisons, but I don't know what the minimum python version target is for this library, since support was added in 3.4.  I see an update referencing 3.8, but I am not sure.

**Comment by chrisruk** [comment]
Hi, I just committed a small patch to remove the invalid device checking from each of the Device subclasses such as Motor, to the Device class itself, which I think makes the code simpler, as we don't have do a check in each subclass anymore.

**Comment by mutesplash** [comment]
Ah!  Yes, I like it much better than my hacky solution.  Can I rework this to keep the descriptions or is that unwelcome?

**Comment by mutesplash** [comment]
Closing this due to @chrisruk 's changes to consolidate device checking making most of this patch obsolete.  I opened a new PR #124 for the device descriptions.

**Pull request closed without merging**